src/main/java/com/pinterest/secor/parser/ChannelDateMessageParser.java

```diff
@@ -135,18 +135,13 @@ private String getPrefix(String prefixIdentifier) {
 
 	private String getChannel(JSONObject jsonObject) {
 		String rawChannelStr = "";
-		Map<String, Object> dimensions = (HashMap<String, Object>) jsonObject.get("dimensions");
-		Map<String, Object> context = (HashMap<String, Object>) jsonObject.get("context");
-
-		String channel = (String) jsonObject.get("channel");
-		if (channel != null && !channel.isEmpty()) {
-			rawChannelStr = channel;
-		} else if (dimensions != null && dimensions.get("channel") != null) {
-			rawChannelStr = (String) dimensions.get("channel");
-		} else if (context != null && context.get("channel") != null) {
-			rawChannelStr = (String) context.get("channel");
-		} else {
-			rawChannelStr = "in.ekstep";
+		Map<String, Object> derivedLocData = (HashMap<String, Object>) jsonObject.get("derivedlocationdata");
+
+		if (derivedLocData != null && derivedLocData.get("state") != null) {
+			rawChannelStr = (String) derivedLocData.get("state");
+		}
+		else {
+			rawChannelStr = "others";
```

Comment:
can we enable the channel identification as a rule?
for ex: Can I configure something like this
message.channel.identifier=event.derivedlocationdata.state
or
message.channel.identifier=event.channel eq '<custodianorgid>'?event.derivedlocationdata.state:event.channel

Reply:
Enabled channel identification as a rule. **message.channel.identifier** - takes list of fields to work for both raw and summary events
Example configs:
`secor.partition.message.channel.identifier=context.channel,dimensions.channel` - channel telemetry field as identifier
`secor.partition.message.channel.identifier=derivedlocationdata.state` - derived loc state as identifier